### DISCUSSION TITLE
Guru Meditation Error: Core 0 panic'ed (Load access fault). Exception was unhandled. (IDFGH-12456)

### DISCUSSION THREAD

**Discussion by GregoryCle** [micropython/micropython#14197] [ESP32]
Hi and thank you in advance for your help.

I have a circuit with an ESP32C3. This works well with already compiled Micropython code. I have already managed to compile my own firmware except that I had a computer problem and I had to redo the entire installation. Since then, it no longer works.

I posted my message on the espressif forum, but I am told that the problem probably comes from Micropython.

I can't find the reason. I think it comes from my IDF and settings. I am attaching sdkconfig as an attachment.

`   
`Rebooting...
ESP-ROM:esp32c3-api1-20210207
Build:Feb 7 2021
rst:0xc (RTC_SW_CPU_RST),boot:0xf (SPI_FAST_FLASH_BOOT)
Saved PC:0x4038061c
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd5820,len:0xf28
load:0x403cc710,len:0x89c
load:0x403ce710,len:0x2b14
entry 0x403cc710
Guru Meditation Error: Core 0 panic'ed (Load access fault). Exception was unhandled.

Core 0 register dump:
MEPC : 0x4004c48c RA : 0x4004c4b0 SP : 0x3fcde2b0 GP : 0x3fc97400
TP : 0x00000000 T0 : 0x4005890e T1 : 0x00000000 T2 : 0x00000000
S0/FP : 0x000000ff S1 : 0x00000050 A0 : 0x000000ff A1 : 0x3fcde2e0
A2 : 0x0012ad40 A3 : 0x03ff0000 A4 : 0x0000e000 A5 : 0x60ffe01c
A6 : 0xfa000000 A7 : 0x00000003 S2 : 0x0000000c S3 : 0x3fcde434
S4 : 0x42000000 S5 : 0x3c150000 S6 : 0xffff0000 S7 : 0x3fcd6000
S8 : 0x00050020 S9 : 0x00010020 S10 : 0x00000006 S11 : 0x3fcde3cc
T3 : 0x60008000 T4 : 0x60008000 T5 : 0xffff01ff T6 : 0x00000000
MSTATUS : 0x00001881 MTVEC : 0x40380001 MCAUSE : 0x00000005 MTVAL : 0x60ffe01c
MHARTID : 0x00000000

Stack memory:
3fcde2b0: 0x0000000c 0x00000050 0x00000001 0x420007e2 0x400391e6 0x00000000 0x00000000 0x60008000
3fcde2d0: 0x00000001 0x000001e0 0x00000006 0x00000050 0x00000001 0x000001e0 0x00000003 0x000000a0
3fcde2f0: 0x0000000c 0x421430b3 0x00000000 0x40380366 0x4038025c 0x421430b3 0x00000001 0x403cf714
3fcde310: 0x66296ba1 0x3a8457bf 0x30470726 0x00143094 0x3f5cb405 0xc8504df6 0xccf84e19 0x00150000
3fcde330: 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 0x00000000 0x00000000 0x3fcdf000
3fcde350: 0x00004308 0x3ff1c14c 0x0000ffff 0xffffffff 0xffffffff 0x3fcde4b0 0xffffffff 0x403cfaf8
3fcde370: 0x80821055 0x80824511 0x00010000 0x001f0000 0x00010000 0x2f0206e9 0x4038025c 0x000000ee
3fcde390: 0x03030005 0x0000c700 0x01000000 0x3c150020 0x0003c3f8 0x3fc96c00 0x00002fdc 0x40380000
3fcde3b0: 0x00000c14 0x42000020 0x00143094 0x40380c14 0x00015fb4 0x50000000 0x00000018 0x00000000
3fcde3d0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde3f0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde410: 0x00000000 0x00000000 0x00000000 0x00010020 0x0004c420 0x0004f404 0x00050020 0x001930bc
3fcde430: 0x001a9078 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde450: 0x00000000 0x00000000 0x00000000 0x001990c0 0x66296ba1 0x3a8457bf 0x30470726 0x769d0c42
3fcde470: 0x3f5cb405 0xc8504df6 0xccf84e19 0x35e94ceb 0x400481b0 0x0000007d 0x00000000 0x3fce0000
3fcde490: 0x00000000 0x0000007d 0x00004310 0x403cc796 0x080c4632 0x0bb386aa 0x00000000 0xffffffff
3fcde4b0: 0x00000000 0x00000000 0x00010000 0x001f0000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde4d0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde4f0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde510: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde530: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde550: 0x600c5200 0x600c5200 0x00000000 0x4004a296 0x069347e1 0xc2020d80 0x3fcde668 0x00000000
3fcde570: 0xffffffff 0x00000000 0x00010000 0x2f0203e9 0x403cc710 0x403ce710 0x00002b14 0x000000ee
3fcde590: 0x03030005 0x0000c700 0x01000000 0x3c004310 0x7533885e 0x7dbb14c9 0x7fa53323 0x42293243
3fcde5b0: 0x91690b1e 0xb4a70271 0xb0434621 0xe85d2c06 0x413a2dcb 0x00000000 0x00000000 0x00000000
3fcde5d0: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x3fcde700 0x00000101 0x00000002
3fcde5f0: 0x7dbb14c9 0x7fa53323 0x42293243 0x91690b1e 0xb4a70271 0xb0434621 0xe85d2c06 0x413a2dcb
3fcde610: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde630: 0x00876713 0x8082dbb8 0x00000000 0x7d000000 0x00000080 0x00000000 0x00000000 0x00000000
3fcde650: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x80180200
3fcde670: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000
3fcde690: 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000

ELF file SHA256:
[sdkconfig.txt](https://github.com/micropython/micropython/files/14773187/sdkconfig.txt)

`

**Comment by LearningToPi** [comment]
Just ran into a similar problem on an ESP32-C3 board.  After multiple attemps of getting the following errors (right before the crash dump):
E (323096) BLE_INIT: hci inits failed
E (323096) BLE_INIT: nimble host init failed

I tried initializing then disabling the WLAN manually on the REPL and saw the following:

E (124786) BLE_INIT: Malloc failed
Funcs table create fails
E (124786) BLE_INIT: esp_bt_controller_init -2
E (124786) BLE_INIT: controller init failed

Led me to believe this is just a memory issue.  I have multiple largish .py files, and after pre-compiling to .mpy files and deleting the .py files from the flash, it booted with no issue!  I only needed to replace 1 .py file (that was 48k) with an .mpy file (that was 19k) to resolve the problem.

If you are carrying over code that worked on a old ESP32, or an ESP32-S*, you may need to shrink the code to get it to run on the ESP32-C3 (and probably other C* chips).

If you haven't used it, you can get mpy-cross as a pypi package, or if you are compiling your own firmware, grab it from the mpy-cross folder in your micropython source code.

There is info on .mpy files in the documentation here:  https://docs.micropython.org/en/latest/reference/mpyfiles.html

**Comment by jkorte-dev** [comment]
I had the same issue with esp32c3. It was a lack of memory and went away by using esp32c6 or mpy files. I think the best solution to reduce memory footprint is using romfs.
